test/test_torch.py

```diff
@@ -16894,6 +16893,51 @@ def test_bmm(self, device, dtype):
             self.assertRaises(RuntimeError, lambda: torch.bmm(b1, b2.cuda()))
             self.assertRaises(RuntimeError, lambda: torch.bmm(b1.cuda(), b2))
 
+    @onlyCUDA
+    @wrapDeterministicFlagAPITest
+    def test_cublas_config_deterministic_error(self, device):
+        test_cases = [
+            # (function, (tensor sizes))
+            (torch.mm, ((2, 2), (2, 2),)),
+            (torch.mv, ((2, 2), (2,),)),
+            (torch.bmm, ((1, 2, 2), (1, 2, 2),))]
+
+        test_configs = [
+            # (config, is deterministic)
+            ('garbage', False),
+            (None, False),
+            (':4096:8', True),
+            (':16:8', True)]
+
+        cublas_var_name = 'CUBLAS_WORKSPACE_CONFIG'
+        is_cuda10_2_or_higher = (
+            (torch.version.cuda is not None)
+            and ([int(x) for x in torch.version.cuda.split(".")] >= [10, 2]))
+
+        processes = []
+        for fn, arg_sizes in test_cases:
+            args = []
+            for arg_size in arg_sizes:
+                args.append(torch.randn(*arg_size, device=device))
+
+            for config, is_config_deterministic in test_configs:
+                if config is None:
+                    if os.environ.get(cublas_var_name) is not None:
+                        del os.environ[cublas_var_name]
+                else:
+                    os.environ[cublas_var_name] = config
+
+                should_throw_error = is_cuda10_2_or_higher and not is_config_deterministic
+                p = torch.multiprocessing.spawn(
```

Comment:
I am not really sure why Windows is failing. However, it would seem to me that it would be a lot less error prone if you just start an honest to goodness subprocess (as is what the other code does that is existing in the test suite). This would also save you from having to fuck around with the *process-global* environment state, and it also prevents weird problems if you are using fork strategy rather than spawn strategy.

Reply:
Yeah I tried out a couple things using the `multiprocessing` module directly (not `torch.multiprocessing`), but seems like it acts a bit too differently on Windows to easily make it work properly. `subprocess` it is.

Reply:
I _think_ my latest changes will fix the test. I manually ran it successfully several times in a row on the CI machine. Then it started failing with `CUDA out of memory` errors. However, I noticed that the CI job was running other CUDA tests at the same time that I was doing manual runs, which was likely the cause of the failures. To be confident that I'm not introducing a flaky test, I'm going to ssh in once the Windows job starts up again, stop the test suite process, and manually run the test a few dozen times.

Reply:
Welp, the test runs fine by itself, but if it's included as part of the whole test suite, it fails with `CUDA out of memory`.